doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,175 @@
+* Easy to use API for plugin implementers
+
+## Dependency Changes
+
+* New: Using Lunchbox plugin API for loading plugins
+
+## API
+
+Change signature of Lunchbox::Plugin::HandlesFunc to:
+    typedef boost::function< bool ( InitDataT& ) > HandlesFunc;
+
+    const Plugins& Lunchbox::PluginFactory<..>::getPlugins();
+
+    class PluginInfo
+    {
+    public:
+        std::string name;
+        float ratio; //!< Normalized 0..1 size after compression
+        float speed; //!< Relative speed compared to RLE compressor
+    };
+    typedef std::vector< PluginInfo > PluginInfos;
+
+    class CompressorPlugin : public Lunchbox::Plugin< Compressor, PluginInfos >
+    {
+    public:
+        std::unique_ptr< Compressor > create( const std::string& name );
+    };
+
+    class Compressor
+    {
+    public:
+        typedef lunchbox::Bufferb Result;
+        typedef std::vector< Result > Results;
+
+        // size[1] == 1 for 1D data
+        virtual const Results& compress( const void* const data,
+                                         const size_t size ) = 0;
+        virtual const Result& decompress( const Results& input,
+                                          const size_t size ) = 0;
+
+        static void registerEngine( const PluginInfo& info,
+                                    const Constructor& ctor );
+    protected:
+        Results compressed;
+        Result uncompressed;
+
+    private:
+        static bool _handles( PluginInfo& info );
+        static bool _create( const std::string& name );
+    };
+
+
+## Examples
+
+pression::Compressor frontend:
+
+    std::string Compressor::choose( const PluginRegistry& registry )
+    {
+        PluginInfo candidate;
+        candidate.quality = 1.0f;
+        candidate.speed = 1.0f;
+        float rating = powf( candidate.speed, .3f ) / candidate.ratio;
+
+        const Plugins& plugins = registry.getPlugins();
+        for( const Plugin* plugin : plugins )
+        {
+            for( const PluginInfo& info : plugin->getInfo( ))
+            {
+                float newRating = powf( info.speed, .3f ) / info.ratio;
+                if( newRating > rating )
+                {
+                    candidate = info;
+                    rating = newRating;
+                }
+            }
+        }
+        return candidate.name;
+    }
+
+    PluginInfos Plugin::getInfos()
+    {
+        PluginInfos infos;
+        if( handles( infos ))
+            return infos;
+        return PluginInfos();
+    }
+
+Compressor implementation:
+
+    #include "zstd/lib/zstd.h"
+
+    namespace pression
+    {
+    namespace plugin
+    {
+    namespace
+    {
+    static PluginInfo _getInfo()
+    {
+    }
+
+    static bool _register()
+    {
+        Compressor::registerEngine( { "pression::CompressorZSTD", .47f, .25f },
+                                    CompressorZSTD::newInstance );
+        return true;
+    }
+
+    static const bool LB_UNUSED _initialized = _register();
```

Comment:
How is all this code is better than:

     lunchbox::PluginRegisterer< CompressorZSTD > registerer;

Static initialization fiasco shouldn't be a concern here as no other code should depend on this at initialization.